src/components/common/DuplicateButton.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+import FileCopyIcon from '@mui/icons-material/FileCopy';
+import { ListItemIcon, MenuItem } from '@mui/material';
+
+import { DiscriminatedItem } from '@graasp/sdk';
+
+import { useBuilderTranslation } from '@/config/i18n';
+import { mutations } from '@/config/queryClient';
+import { ITEM_MENU_DUPLICATE_BUTTON_CLASS } from '@/config/selectors';
+import { BUILDER } from '@/langs/constants';
+import { getParentsIdsFromPath } from '@/utils/item';
+
+const DuplicateButton = ({
+  item,
+}: {
+  item: DiscriminatedItem;
+}): JSX.Element => {
+  const { mutate: copyItems } = mutations.useCopyItems();
+  const { t: translateBuilder } = useBuilderTranslation();
+
+  const handleDuplicate = () => {
+    const parentsIds = getParentsIdsFromPath(item.path);
+    // get the close parent if not then undefined
+    const to =
+      parentsIds.length > 1 ? parentsIds[parentsIds.length - 2] : undefined;
```

Comment:
I think the idea here was to get the parent from the path. I changed it a bit but the idea is the same: you get the latest (or before latest since latest is yourself) parent to copy into.